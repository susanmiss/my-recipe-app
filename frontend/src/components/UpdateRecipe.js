import React, {useState, useEffect} from 'react';
import axios from 'axios';




const UpdateRecipe = (props) => {

    const [state, setState] = useState({
        recipeTitle: '',
        recipeContent: '',
        slug: ''
    })

    

    const {recipeTitle, recipeContent, slug} = state

 
        useEffect(() => {
            axios.get(`${process.env.REACT_APP_API}/${props.match.params.slug}`)
            .then(response => {
                const {recipeTitle, recipeContent, slug} = response.data;
                setState({...state, recipeTitle, recipeContent, slug}) 
            }) 
            .catch(error => alert('Error loading single recipe'))
        }, []);  

        const handleChange = (name) => (event) => {          
            setState({...state, [name]: event.target.value}) 
        }

            const handleSubmit = event => {
                event.preventDefault()  
                axios
                .put(`${process.env.REACT_APP_API}/${slug}`, {recipeTitle,recipeContent})
                .then(response => {
                    console.log(response)  
                    const {recipeTitle, recipeContent, slug} =  response.data      
                    setState({...state, recipeTitle, recipeContent})
                    alert(`Recipe titled ${recipeTitle} is updated.`)
                })
                .catch(error => {
                    console.log(error.response)
                    alert(error.response.data.error);
                }) 
        }  


        const showUpdateForm = () => (
            <form onSubmit={handleSubmit}>
    
                  <div className="form-group" >
                      <label className="text-muted">Recipe title</label>
                      <input onChange={handleChange('recipeTitle')} value={recipeTitle} type="text" className="form-control" placeholder="Recipe Title" required />
                  </div>
              
                  <div className="form-group" >
                      <label className="text-muted">Recipe</label>
                      <input onChange={handleChange('recipeContent')} value={recipeContent} type="text" className="form-control" placeholder="Recipe" required />
                  </div>
          
                  <div>
                      <button className="primary">Update Recipe</button>
                  </div>
            </form>      
        )
          



    return (
        <div className="container pb-5">
           {/* {JSON.stringify(props)} */}

          <br />
              <h1>Update Recipe</h1>
               {showUpdateForm()}
        </div>
        )
}

export default UpdateRecipe;
