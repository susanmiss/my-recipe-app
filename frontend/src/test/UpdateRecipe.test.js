import React from 'react';
import ReactDOM from 'react-dom'
import {shallow} from 'enzyme'; 
import renderer from 'react-test-renderer';
import UpdateRecipe from '../components/UpdateRecipe'


describe("<UpdateRecipe />", () => {
    
 
  it('renders correctly', () => {
      const tree = renderer
          .create(<UpdateRecipe />)
          .toJSON();
      expect(tree).toMatchSnapshot();
  });
  let wrapper;
  beforeEach(() => wrapper = shallow(<UpdateRecipe />));
  
  it('Renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
  })
  
  it('should render a <div />', () => {
      expect(wrapper.find('div').length).toEqual(4);
  });


 
})