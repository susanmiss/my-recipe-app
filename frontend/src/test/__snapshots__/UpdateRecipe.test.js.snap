// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpdateRecipe /> Renders correctly 1`] = `ShallowWrapper {}`;

exports[`<UpdateRecipe /> renders correctly 1`] = `
<div
  className="container pb-5"
>
  {}
  <br />
  <h1>
    Update Recipe
  </h1>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        className="text-muted"
      >
        Recipe title
      </label>
      <input
        className="form-control"
        onChange={[Function]}
        placeholder="Recipe Title"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="text-muted"
      >
        Recipe
      </label>
      <input
        className="form-control"
        onChange={[Function]}
        placeholder="Recipe"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div>
      <button
        className="primary"
      >
        Update Recipe
      </button>
    </div>
  </form>
</div>
`;
